spin/cmds/meson.py

```diff
@@ -307,12 +310,12 @@ def build(
             )
         return
 
-    meson_args = list(meson_args)
+    meson_args_setup = list(meson_args.get("setup", tuple()))
 
     if gcov:
-        meson_args = meson_args + ["-Db_coverage=true"]
+        meson_args_setup = meson_args_setup + ["-Db_coverage=true"]
 
-    setup_cmd = _meson_cli() + ["setup", build_dir, "--prefix=/usr"] + meson_args
+    setup_cmd = _meson_cli() + ["setup", abs_build_dir, "--prefix={}".format(abs_install_dir)] + meson_args_setup
```

Comment:
I think I'd prefer approach in my PR where it's possible to explicitly set `prefix`, I don't think that's possible here.

I'd also really like it if it were possible for downstream projects to adapt to changes in commands without explicitly updating their own spin code.

Reply:
Makes sense to me. Basically the changes I made work for both SciPy and NumPy as well. The changes in `main` branch of `spin` work only with NumPy and not with SciPy. Your change is better (it is backwards compatible), however since `--prefix=/usr` doesn't work with SciPy, so we will have to pass `--prefix=<abs_install_dir>` in SciPy project. 

Basically the choice to be made here is,

1. Option 1 - Follow https://github.com/scientific-python/spin/pull/241/files and pass `--prefix=<abs_build_dir>`.
2. Option 2 - Follow this PR (once it's merged in `main` branch of `spin`). Then NumPy and SciPy will work fine as is. However, its hard to say if other projects using `spin` will break or not.

I am fine with Option 1. That way only SciPy interface will need changes and no other project using spin will break.

cc: @rgommers @stefanv

Reply:
SciPy has the same constraints as NumPy. It doesn't matter what `--prefix` is set to, I'd expect things to work for both project. The actual install location is controlled by `--destdir`; `--prefix` is turned into a relative path under `destdir`. Look in `numpy/build-install/` to see what I mean (the next subdir there is `usr`).

Reply:
Another `--prefix` patch landed in `main` already. Did you try dropping this change @czgdp1807?